packages/common-integration-tests/src/leaderboard-client.ts

```diff
@@ -0,0 +1,1062 @@
+    });
+
+    it('validates the leaderboard name', async () => {
+      const elements = new Map([
+        [123n, 100],
+        [456n, 200],
+        [789n, 300],
+      ]);
+      const response = await leaderboardClient.leaderboardUpsert(
+        integrationTestCacheName,
+        '   ',
+        elements
+      );
+      expectWithMessage(() => {
+        expect(response).toBeInstanceOf(LeaderboardUpsert.Error);
+      }, `expected ERROR but got ${response.toString()}`);
+      const responseError = response as LeaderboardUpsert.Error;
+      expectWithMessage(() => {
+        expect(responseError.errorCode()).toEqual(
+          MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        );
+      }, `expected INVALID_ARGUMENT_ERROR but got ${responseError.errorCode()} ${responseError.message()}`);
+    });
+
+    it('validates the number of elements', async () => {
+      const elements = new Map<bigint, number>();
+      const response = await leaderboardClient.leaderboardUpsert(
+        integrationTestCacheName,
+        leaderboardName,
+        elements
+      );
+      expectWithMessage(() => {
+        expect(response).toBeInstanceOf(LeaderboardUpsert.Error);
+      }, `expected ERROR but got ${response.toString()}`);
+      const responseError = response as LeaderboardUpsert.Error;
+      expectWithMessage(() => {
+        expect(responseError.errorCode()).toEqual(
+          MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        );
+      }, `expected INVALID_ARGUMENT_ERROR but got ${responseError.errorCode()} ${responseError.message()}`);
+    });
+
+    it('creates new leaderboard to insert elements into', async () => {
+      const elements = new Map([
+        [123n, 100.0],
+        [456n, 200.0],
+        [789n, 300.0],
+      ]);
+      const response = await leaderboardClient.leaderboardUpsert(
+        integrationTestCacheName,
+        leaderboardName,
+        elements
+      );
+      expectWithMessage(() => {
+        expect(response).toBeInstanceOf(LeaderboardUpsert.Success);
+      }, `expected SUCCESS but got ${response.toString()}`);
+
+      const fetchResponse = await leaderboardClient.leaderboardFetchByRank(
+        integrationTestCacheName,
+        leaderboardName
+      );
+      expectWithMessage(() => {
+        expect(fetchResponse).toBeInstanceOf(LeaderboardFetch.Found);
+      }, `expected HIT but got ${fetchResponse.toString()}`);
+      const receivedElements = (
+        fetchResponse as LeaderboardFetch.Found
+      ).valueArray();
+      const expectedElements = [
+        {
+          id: 123n,
+          score: 100,
+          rank: 0n,
+        },
+        {
+          id: 456n,
+          score: 200,
+          rank: 1n,
+        },
+        {
+          id: 789n,
+          score: 300,
+          rank: 2n,
+        },
+      ];
+      expect(receivedElements).toEqual(expectedElements);
+    });
+
+    it('inserts new elements into an existing leaderboard', async () => {
```

Comment:
in general, you want any of these `it` blocks to be a self-contained test that can be run independently. it looks like you are assuming some previous state in this one. not a blocker for this PR but a good thing to fast-follow on and a good best practice for future tests.